src/transformStateWithClones.js

```diff
@@ -7,7 +7,67 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      if (arr.length === 0) {
+        const obj = Object.assign({}, state);
+
+        for (const prop in obj) {
+          delete obj[prop];
+        }
+
+        arr.push(obj);
+      } else {
+        const obj = Object.assign({}, arr[arr.length - 1]);
+
+        for (const prop in obj) {
+          delete obj[prop];
+        }
+
+        arr.push(obj);
+      }
```

Comment:
why do you need this` if else` construction? just make a copy of the `state` and push the current state in the end of every interation